pkg/services/ngalert/api/api_testing.go

```diff
@@ -43,22 +43,23 @@ type TestingApiSrv struct {
 	featureManager  featuremgmt.FeatureToggles
 	appUrl          *url.URL
 	tracer          tracing.Tracer
+	rulesStore      RuleStore
 }
 
 // RouteTestGrafanaRuleConfig returns a list of potential alerts for a given rule configuration. This is intended to be
 // as true as possible to what would be generated by the ruler except that the resulting alerts are not filtered to
 // only Resolved / Firing and ready to send.
 func (srv TestingApiSrv) RouteTestGrafanaRuleConfig(c *contextmodel.ReqContext, body apimodels.PostableExtendedRuleNodeExtended) response.Response {
+	folder, err := srv.rulesStore.GetNamespaceByUID(c.Req.Context(), body.NamespaceUID, c.OrgID, c.SignedInUser)
+	if err != nil {
+		return toNamespaceErrorResponse(dashboards.ErrFolderAccessDenied)
```

Comment:
Unfortunately, this becomes mandatory in this API. Otherwise, we risk providing incorrect routing information. Previously we took the folder title from the request and it was enough but UI cannot provide full path in the folder title anymore. So, I had to update it. Anyway, as long as user has access to the rule it has access to the folder it contains.